infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java

```diff
@@ -0,0 +1,118 @@
+package com.redhat.service.smartevents.infra.v2.api.models.dto;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public abstract class BaseResourceDTO {
+    @JsonProperty("id")
+    private String id;
+
+    @JsonProperty("name")
+    private String name;
+
+    @JsonProperty("customerId")
+    private String customerId;
+
+    @JsonProperty("owner")
+    private String owner;
+
+    @JsonProperty("operation")
+    private OperationType operationType;
+
+    @JsonProperty("generation")
+    private long generation;
+
+    @JsonProperty("expiry")
+    private ZonedDateTime expiry;
```

Comment:
@manstis I think it is probably better for us to communicate the timeout period in seconds and let the data plane calculate when it's deadline for deployment is. So essentially the DP is given `X` seconds to deploy something from the first time it sees the resource from the CP.

I'd also say something like `deploymentDeadline` or `readyDeadline` is a bit more explanatory as to the intent of the field here.

Reply:
OK, changed to a property called `timeout` instead (being the number of seconds).

`Deadline` suggests a timestamp to me.